tests/components/ecobee/test_climate.py

```diff
@@ -335,3 +337,17 @@ async def test_set_fan_mode_auto(thermostat, data):
     data.ecobee.set_fan_mode.assert_has_calls(
         [mock.call(1, "auto", "nextTransition", holdHours=None)]
     )
+
+
+async def test_turn_aux_heat_on(thermostat, data):
+    """Test when aux heat is set on.  This must change the HVAC mode."""
+    data.reset_mock()
+    thermostat.turn_aux_heat_on()
+    data.ecobee.set_hvac_mode.assert_has_calls([mock.call(1, "auxHeatOnly")])
+
+
+async def test_turn_aux_heat_off(thermostat, data):
+    """Test when aux heat is tuned off.  Must change HVAC mode back to last used."""
+    data.reset_mock()
+    thermostat.turn_aux_heat_off()
+    data.ecobee.set_hvac_mode.assert_has_calls([mock.call(1, "auto")])
```

Comment:
You could maybe validate that `self._last_active_hvac_mode` has been applied

Reply:
good thought.. but honestly I'm not sure how to test that functionality.  There do not appear to be any current tests related to `_last_active_hvac_mode`.  It is only set `async_update` which probably wouldn't make sense to call in a mocked test.  I tried manually setting it, but it didnt work.

LMK if you have thoughts

Reply:
good call :+1: fixed